### DISCUSSION TITLE
Integration across different conditions of multiomic seurat objects

### DISCUSSION THREAD

**Discussion by Onero23** [satijalab/seurat#4015] [Q&A] [answered]
First of all thanks a lot for Seurat!
I was wondering what's the best way to integrate datasets of different conditions (patients, control) of multiomic seurat objects (scATAC&scRNA). Do I have to start to integrate the scATAC and scRNA data first for each Seurat first and then try to integrate those? Will there be a solution in future Seurat versions?
Thank you very much!
Wolfgang

**Comment by timoast** [comment, marked as answer]
If you have multiome data there's no need to integrate the RNA and ATAC assays as these are from the same cells. To integrate across batches, I'd suggest performing the standard integration workflow using the RNA assay from each dataset.

**Reply by Onero23** [reply]
Thanks a lot Tim! By the way thank you very much for Signac! It's awesome! 
So to get the scATAC information in a UMAP/tSNE plot in a multiomic approach is it ok to use FindMultiModalNeighbors of the integrated RNA data and the ATAC data and to run UMAP on the created wknn?
Thank you very much!

**Reply by ahepperla** [reply]
Thanks Tim! So no need to worry about scATAC integration between samples as long as the scRNA is integrated?

**Reply by timoast** [reply]
@ahepperla you could also integrate the LSI embeddings using the anchors identified using the RNA assay. For example:

```
integrated <- IntegrateEmbeddings(anchorset = anchors, reductions = object[["lsi"]], ...)
```

You'll need to first merge all the ATAC assays together and compute a shared LSI, which would then be corrected in the `IntegrateEmbeddings` step.

**Reply by melissansf** [reply]
Hi! Jumping in on this.. could I clarify if we then follow the vignette on merging scATACseq objects in Signac? To also ensure that a common set of peaks that go into each fragment file for each multiome object, before finding the integration anchors on the SCT assay and integrating them. After integration, if I follow your comment correctly, we would then compute a shared LSI and integrate the embeddings?

Also thanks very much for the awesome vignettes on Signac and Seurat!

**Comment by satijalab** [comment]
Just to add some further context to this, we will be introducing the opportunity to identify anchors (and therefore integrate/transfer data across multi-omic datasets) in future versions of Seurat, hopefully soon. For now, we would recommend integrating on the basis of transcriptomic measurements.

**Reply by mzz92** [reply]
Wanted to ping this thread and see if multi-omic dataset integration is possible with Seurat 4? I have several 10X Multiome datasets and have performed the WNN analysis on each. Is it possible to integrate all of the Multiome datasets to use all of the RNA and ATAC data?

Looking at the vignettes, FindMultiModalNeighbors does not accept a list for object, and FindIntegrationAnchors only seems to integrate on RNA or on ATAC alone.

BTW--fantastic, invaluable resource.

**Reply by JoreVW** [reply]
I would like to jump on this as well. I'm starting to analyse my first 10X multiome datasets and I was indeed wondering if with Seurat 4 the integration of the datasets (control vs ko) is possible now? Or should I integrate RNA data, integrate ATAC, add them and than preform the WNN analysis? 
Thank you indeed for both Signac and Seurat!

**Reply by qimiaonnegliguo** [reply]
> I would like to jump on this as well. I'm starting to analyse my first 10X multiome datasets and I was indeed wondering if with Seurat 4 the integration of the datasets (control vs ko) is possible now? Or should I integrate RNA data, integrate ATAC, add them and than preform the WNN analysis? Thank you indeed for both Signac and Seurat!

Hey Jore, may I ask your solutions? Is it possible and reasonable to 'integrate RNA data, integrate ATAC, add them and than preform the WNN analysis'? Thanks a lot : D

**Reply by guokai8** [reply]
Hi @JoreVW,
Could you share your code and I am not sure if I understand correctly for the details. Thanks!
K

**Reply by JoreVW** [reply]
Hi @guokai8, sorry I deleted my comment because this was not the correct approach for me. For the multiomics data set you should not split the RNA and ATAC, it's from the same cell so it does not make sense to integrate them again. So in my case, I did QC on my sample for both RNA and ATAC where both assays are within the same object. Next I merged and integrated my datasets (WT and KO in my case) based on the anchors. This is in my modest opinion the correct way to analyse and merge 2 multiome datasets. Let me know if this helps.

**Reply by guokai8** [reply]
Hi @JoreVW,
Thanks for your suggestions. I had few questions here: 1. after you did QC you will have exactly the same cell numbers in both ATAC and RNAseq data, right? 2. before you construct your object, did you merge your ATAC data based on the merge manual in signac (https://stuartlab.org/signac/articles/merging.html)? 3. you will use the cell type annotation from scRNAseq for ATACseq.
Thanks again!
K

**Reply by JoreVW** [reply]
Hi @guokai8, to answer your questions,

1. yes, indeed you end up with the same number of cells
2. I noticed when I first merged my RNA data, and next my ATAC, that I lost my integrated_RNA reduction. Therefore I first merged the objects based on the RNA to get "Integrated_Object", next I merged the ATAC based on the RNA-anchors but created a new object after this merge "Integrated_ATAC", since you lose the integrated RNA reduction. Next I copied the integrated LSI reduction from the Integrated_ATAC into my original object where the RNA was already merged. The code is straightforward : Integrated_Object@reductions$integratedLSI <- Integrated_ATAC@reductions$integratedLSI
3. Yes, I used the cell type annotations form the RNA because the quality of my RNA sequencing was better and gave me more accurate annotations. 

Just to be sure we are discussing the same experiment, by merging you mean merging 2 of 2 objects with both RNA and ATAC data, like merging 2 condition like wildtype and knockout? You do not mean merging an ATAC experiment with an RNA experiment?

Jore

**Reply by guokai8** [reply]
Hi @JoreVW,
Thanks for your classification. I first merge data from ATACseq from both WT and KO, and also merge scRNAseq data from both WT and KO.  So now we just need to integrate the scRNAseq data and then use the anchors from scRNAseq data, right?
Thanks!
K
